doc/conf.py

```diff
@@ -41,9 +42,10 @@
 # necessary when building the sphinx gallery
 pyvista.BUILDING_GALLERY = True
 
-# SG warnings
-import warnings
+# ignore any EGLWarnings when building documentation with VTK compiled with OSMesa
+os.environ['PYTHONWARNINGS'] = 'EGLWarning'
```

Comment:
1. Testing locally, none of these actually suppress the warning:
```bash
$ PYTHONWARNINGS=EGLWarning python
$ PYTHONWARNINGS='ignore::EGLWarning' python
$ PYTHONWARNINGS='ignore::pyvista.plotting.renderer.EGLWarning' python
```
Not sure what the correct syntax is here.

2. If we do this, we should let any pre-defined warning settings through, so we should probably check if it's empty and append with a comma if something's already in this envvar.

Reply:
1. I think This is because the current implementation overwrites the env var with an incorrect string.

Reply:
@MatthewFlamm sorry for the ambiguity: I'm just opening up a python REPL with that envvar and running
```py
import pyvista
pl = pyvista.Plotter()
pl.enable_anti_aliasing()
_ = pl.add_mesh(pyvista.Sphere(), show_edges=True)
pl.show()
```
from the `enable_anti_aliasing()` doctest. Since the wrong envvar is only in `doc/conf.py` it should not interfere here.

Reply:
Hmmm yes.  It seems that `PYTHON WARNINGS` cannot handle custom warning classes.  See https://nedbatchelder.com/blog/201810/why_warnings_is_mysterious.html as another example.  Can we ignore based on message then?

Reply:
Ugh, good find. I've been looking everywhere in the documentation and PEPs, with the growing suspicion that we can only use built-in names on the command line :/ Inconvenient.

In this case I agree, we could probably only filter for message, short of coming up with a different kind of hack that shoehorns proper `warning` module API calls in subprocesses... I mean it would only take defining an envvar of our own, which would trigger setting up a filter in `pyvista/__init__.py`... :laughing: